src/python/pants/help/help_printer.py

```diff
@@ -427,6 +427,11 @@ def print_cmd(args: str, desc: str):
         print_cmd("help tools", "List all external tools.")
         print_cmd("help backends", "List all available backends.")
         print_cmd("help-advanced backends", "List all backends, including experimental/preview.")
+        print_cmd("help symbols", "List available BUILD file symbols.")
+        print_cmd(
```

Comment:
Why do we have this separation? I would have assumed that 90% of the time the relevant symbols are target types, so what are non-advanced symbol help?

Reply:
we already have `pants help targets` which does that (hence not included by default to not clutter the space here, but still available for completeness if asked for). This is for all the rest, such as custom macros, the `env` and `build_file_dir` etc..

I encourage you to try it ;)

```
BUILD file symbols
------------------

PANTS_VERSION

__defaults__            Provide default field values.

__dependencies_rules__  Declare dependencies rules.

__dependents_rules__    Declare dependents rules.

build_file_dir          Returns the path to the directory of the current BUILD file.

duplicate_rule          DeployJarDuplicateRule(pattern: 'str', action: 'str')

env                     Reference environment variable.

http_source             http_source(url: 'str', *, len: 'int', sha256: 'str', filename: 'str' = '')

node_build_script       A build script, mapped from the `scripts` section of a package.json file.

node_test_script        The test script for this package, mapped from the `scripts` section of a package.json file. The
                        pointed to script should accept a variadic number of ([ARG]...) path arguments.

parametrize             A builtin function/dataclass that can be used to parametrize Targets.

per_platform            An object containing differing homogeneous platform-dependent values.

python_artifact         Represents a Python setup.py-based project.

setup_py                Represents a Python setup.py-based project.

shading_keep            JvmShadingKeepRule(pattern: 'str')

shading_relocate        JvmShadingRelocateRule(package: 'str', into: 'str | None' = None)

shading_rename          JvmShadingRenameRule(pattern: 'str', replacement: 'str')

shading_zap             JvmShadingZapRule(pattern: 'str')
```